### ISSUE TITLE
No "Migrate packages.config to PackageReference"

### ISSUE THREAD

**Issue by pantonis** [NuGet/docs.microsoft.com-nuget#905]
Cannot see "Migrate packages.config to PackageReference" option in VS2017 15.6.7

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 75e25f56-15c8-79d7-5e55-d145b7c1159c
* Version Independent ID: 06388a58-df68-9438-92d7-17359196f5c8
* Content: [Migrating from package.config to PackageReference formats](https://docs.microsoft.com/en-us/nuget/reference/migrate-packages-config-to-package-reference)
* Content Source: [docs/reference/migrate-packages-config-to-package-reference.md](https://github.com/NuGet/docs.microsoft.com-nuget/blob/master/docs/reference/migrate-packages-config-to-package-reference.md)
* Service: **unspecified**
* Product: **nuget**
* GitHub Login: @karann-msft
* Microsoft Alias: **karann**

**Comment by hannesnisula** [comment]
"Visual Studio 2017 Version 15.7 Preview 3 and later"
Although, I thought that the released 15.7 would then also have this migration feature but my 15.7.2 is missing so I downloaded the latest preview to use it (15.8.0).

**Comment by pantonis** [comment]
Dont want to install Preview version. any other way of installing it as an extension?

**Comment by kuenzign** [comment]
If you go to Tools -> Options -> NuGet Package Manager -> General,  You can change the "Default package management format" to "PackageReference".  This should make that feature available to you.  I had trouble finding this as well.

**Comment by rohit21agrawal** [comment]
@pantonis  please note that you don't need to set PackageReference as default format for this feature to become available. The feature was released in 15.7 so it became available to you after you updated your VS from 15.6.7 to 15.7

**Comment by MichaelRumpler** [comment]
@rohit21agrawal I use 15.8 and I also didn't see the option until I set the default format.

**Comment by kuenzign** [comment]
I have 15.7.3 and I can confirm that the option to migrate is not available until you set the default format.  This needs to be in the instructions.

**Comment by cremor** [comment]
@kuenzign I can confirm what @rohit21agrawal wrote, I don't need to change the default to get the menu entry.

Maybe you are running in the known issue that is documented on the page? Because that one is also "fixed" by just opening the NuGet package manager node in the options windows (you don't actually have to change the default setting, just open the settings page and close the window again).

**Comment by topgenorth** [comment]
I will confirm what @rohit21agrawal  said. I have 15.7.5 installed, and **Migrate packages.config to PackageReference** does not show up until I set the default package format to be PackageReference.

**Comment by JeremyNull** [comment]
This shouldn't be closed.  I'm still having this issue (had it on all version of 15.7.x and now 15.8).  I still cannot migrate because I have not right-click context menu for it.

**Comment by sghobbs-ic** [comment]
@JeremyNull I am encountering the same issue. This is apparently still an active known issue, per the PackageReference migration docs: https://docs.microsoft.com/en-us/nuget/reference/migrate-packages-config-to-package-reference

> **Known Issues**
> The Migrate packages.config to PackageReference... option is not available in the right-click context menu
> **Issue**
> When a project is first opened, NuGet may not have initialized until a NuGet operation is performed. This causes the migration option to not show up in the right-click context menu on packages.config or References.
> 
> **Workaround**
> Perform any one of the following NuGet actions:
> 
> - Open the Package Manager UI - Right-click on References and select Manage NuGet Packages...
> - Open the Package Manager Console - From Tools > NuGet Package Manager, select Package Manager Console
> - Run NuGet restore - Right-click on the solution node in the Solution Explorer and select Restore NuGet Packages
> - Build the project which also triggers NuGet restore
> You should now be able to see the migration option. Note that this option is not supported and will not show up for ASP.NET and C++ project types.

**tl;dr:** Nuget tools may be lazy loaded by VS. You need to trigger the nuget tools to load via some nuget action, and the context option should appear.

For me, setting the default package option to 'PackageReference' works, but only in the active VS instance session. If I close VS, I have to do that again to get the option to appear again.

**Comment by jcbeck37** [comment]
I'm using Visual Studio Professional 2017 Version 15.8.6 (I just updated from 15.8.1 to see if it fixed my issue.)

I do not see the menu item when I right-click References. The following things did not make it appear:
- Changing the default package type
- Right-clicking Solution file and restoring NuGet
- Opening the Package Manager UI
- Build the package
- Open the Package Manager Console

**Comment by zivkan** [comment]
@jcbeck37 What kind of project are you trying to migrate? As per the docs, it's not available for ASP.NET projects.

**Comment by seangwright** [comment]
If you are finding that the context menu is not available to migrate an ASP.NET project to Package Reference (understandably, because there can be issues) you can use this VS Extension https://marketplace.visualstudio.com/items?itemName=CloudNimble.NuGetPackageReferenceUpgrader

I have used it successfully on many ASP.NET Web Forms apps, converting from packages.config to Package Reference.

**Comment by babelshift** [comment]
Note that the extension mentioned by @seangwright doesn't work with VS2019.

**Comment by callanac** [comment]
I'm having this issue with VS2019 now @babelshift  , can't believe MS haven't solved this. Seems like it's been a problem since 2018. I've tried everything listed above.

**Comment by markjamesm** [comment]
I'm currently experiencing this issue in Visual Studio 2019 Community Edition (Version 16.8.4) while trying to migrate [a library](https://github.com/vipoo/iRacingSDK.Net) to .NET 5.0 from .NET Framework 4.72. I tried:

* Open the Package Manager UI - Right-click on References and select Manage NuGet Packages...
* Run NuGet restore - Right-click on the solution node in the Solution Explorer and select Restore NuGet Packages

As a last ditch effort, I also tried:

* Go to Tools -> Options -> NuGet Package Manager -> General, Change the "Default package management format" to "PackageReference".

Right clicking the project reference after performing these steps still doesn't show the migration option.

**Comment by zivkan** [comment]
@markjamesm NuGet's "packages.config to PackageReference migration tool" does not change [the project "format"](https://docs.microsoft.com/en-us/nuget/resources/check-project-format), just the package reference format. Even using NuGet's PackageReference migrator, the project would be limited to .NET Framework target frameworks. While .NET 5 has dropped the "Core" from its marketing name, the `TargetFrameworkIdenfier` is still `.NETCoreApp`, just like all other .NET Core target frameworks. .NETCoreApp projects can only use the new SDK style project format, not the older project format that the repo you linked is currently using.

Therefore, in order to target .NET 5, you will need to migrate to the new SDK style project files.

There is a [try-convert](https://github.com/dotnet/try-convert) tool, which you can try, but [take note of the tool's support policy](https://github.com/dotnet/try-convert#support-policy). In my experience, simple class libraries can be con
